feat(runtimes): build and publish PHP 8.1 docker images

shouze opened the pull request:
:whale: will allow to test locally bref with php 8.1 when the docker images will get published :tada: 
:point_right: Related to https://github.com/brefphp/bref/pull/980/files#r695552502
:warning: For now xdebug & blackfire are not included because no php 8.1 version available yet.

<!--
Please explain the motivation behind the changes.

In other words, explain **WHY** instead of **WHAT**.
-->

shouze commented on runtime/layers/fpm-dev/Dockerfile:
```diff
@@ -13,9 +13,8 @@ USER root
 RUN cp /tmp/*.so $(php -r "echo ini_get('extension_dir');")
 
-FROM bref/php-${PHP_VERSION}-fpm
+FROM bref/php-${PHP_VERSION}-fpm as without_extensions
```
Will help for other new php versions upgrade as most of the time blackfire or xdebug are not ready during beta releases.

mnapoli commented on runtime/layers/fpm-dev/Dockerfile:
```diff
@@ -29,3 +28,7 @@ ENV PHP_INI_SCAN_DIR="/opt/bref/etc/php/conf.d:/var/task/php/conf.d:/var/task/ph
 # opcache.validate_timestamps=1 : cancels the flag in the base configuration so that files are reloaded
 CMD /opt/bin/php-fpm --nodaemonize --fpm-config /opt/bref/etc/php-fpm.conf -d opcache.validate_timestamps=1 --force-stderr
+
+FROM without_extensions
```
Why the new `FROM` here? We would lose the EXPOSE, ENV and everything defined above?

shouze replied:
I've checked, `ENV`, `EXPOSE`, `CMD` & `ENTRYPOINT` will be inherited from the above, the only difference is that we won't copy extensions (xdebug, blackfire).

mnapoli replied:
Ohh I think I see it now, thank you!

mnapoli commented:
Thanks! I added a minor feedback, and it seems the build is not running in GitHub Actions

shouze commented:
> Thanks! I added a minor feedback, and it seems the build is not running in GitHub Actions

Thanks, everything should be good now. Seems that some versions of docker build previous targets so I moved without_extensions one at first position and fixed the CI

mnapoli commented:
Thank you!

Just to be sure: no need for new layers here, right?

shouze commented:
> Thank you!
> 
> 
> 
> Just to be sure: no need for new layers here, right?

New layers have been published in previous PR so no, only need to publish docker images.

shouze commented:
BTW a new tag should be required regarding GitHub action
https://github.com/brefphp/bref/blob/master/.github/workflows/deploy-docker.yml

mnapoli commented:
👍  the new release should have done it by now

shouze commented:
>  the new release should have done it by now

Thanks a lot... but I have bad news, an upfix seems necessary https://github.com/brefphp/bref/pull/1008

mnapoli commented:
For the record I'm re-enabling the extensions in #1114
